### ISSUE TITLE
startDebugging method in vscode still return true even when the build fails

### ISSUE THREAD

**Issue by rodrigovaras** [Microsoft/vscode#54214]
Live share extension on the host side is using the startDebugging method to launch a debug configuration remotely. As part of step it will eventually need to invoke a task to build that could fail, the problem is that the API will not report 'false' value during the invocation.
Extract of the code here:

const result = await vscode.debug.startDebugging(folders ? folders[0] : undefined, debugConfiguration);
                        if (!result) {
// THIS code almost never get invoked....
                            throw new Error(`Failed to launch debug configuration:${debugConfiguration.name}`);
                        }

we would like a better mechanism to know what really happen in our launch invocation.
Maybe knowing:
- If the build task was cancelled
- task failed
- Debug configuration is invalid
- Failure to launch

Adding a new richer method may be better...

**Comment by Eskibear** [comment]
I have exact the same problem with the API `vscode.debug.startDebugging`.


When was starting a Java debug session with the API `vscode.debug.startDebugging`, build error occurs. It popped up a message dialog letting me choose whether to "Proceed" or "Abort". This part is provided in extension "Debugger for Java". See: https://github.com/Microsoft/vscode-java-debug/blob/0e86ed2cc7e67f423a25421627dad33a89f4a2a2/src/configurationProvider.ts#L116-L119

After clicking "Abort", `resolveDebugConfiguration` returns "undefined".
According to the [VS Code API](https://github.com/Microsoft/vscode/blob/80d6fa0b8eb5f4c0959b9e2925ec7fd28692880f/src/vs/vscode.d.ts#L8133) , the debug session will not start. 
> Returning the value 'undefined' prevents the debug session from starting.


But actually `vscode.debug.startDebugging` still resolves "true", it does not match the [API spec](https://github.com/Microsoft/vscode/blob/80d6fa0b8eb5f4c0959b9e2925ec7fd28692880f/src/vs/vscode.d.ts#L8306): 

> @return A thenable that resolves when debugging could be successfully started.

**Comment by weinand** [comment]
duplicate of #52737

**Comment by weinand** [comment]
After #53874 has been fixed it now becomes clear that there is another issue that prevents build errors from bubbling up to the extension API.

**Comment by roblourens** [comment]
Is it enough to verify that a build error reaches the startDebugging API? Or was there a more specific issue with it?

**Comment by isidorn** [comment]
The startDebugging API was always returning true. To verify:
1. Have a wrong launch configuration, verify that startDebuggin returns false
2. Have a prelaunch task that fails and verify that startDebugging returns false if you decided not to continue debugging
